### DISCUSSION TITLE
Lights don't work until reset to prefab

### DISCUSSION THREAD

**Discussion by dishfish** [FakeFishGames/Barotrauma#13859] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

在更新之后，潜艇内部的光学组件无法正常照明。在潜艇编辑器内可以看到潜艇的光学组件是处于正常设定状态。在潜艇编辑器中，我发现要点击光学组件，然后点击编辑里面的（重置为预装），然后才会恢复照明功能，在版本更新前不存在这个问题，所以我的潜艇现在一片黑暗无法使用。XD

### Reproduction steps

![3247c48eff855d98ae6088944ed42a95](https://github.com/FakeFishGames/Barotrauma/assets/168256021/fa5035ef-4ef7-4501-a7ce-c5708d39a986)
![2c42fa4480400b01db2b15eca31bf2da](https://github.com/FakeFishGames/Barotrauma/assets/168256021/6cfe798e-cd0c-44c2-b926-ab81ba80f5be)
![1d0ecd460a4ba83a82660e27f0fbdb06](https://github.com/FakeFishGames/Barotrauma/assets/168256021/f537ea12-156b-4438-8a35-10fd1c214b1c)

![a363ca883acb96511cd11644f03cc298](https://github.com/FakeFishGames/Barotrauma/assets/168256021/836b58b1-c9a9-4373-b145-d3b857dc73fa)
如您所见，必须在潜艇编辑器重新设定光学组件才能正常照明，但在游戏中无法重新设定，所以潜艇目前一片黑暗XD

### Bug prevalence

Just once

### Single player or multiplayer?

Single player

### Version

v1.4.5.0 (Blood in the Water Update, hotfix 1)

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
Thank you for the report! 

Unfortunately no-one in the team speaks Chinese, but I think I understood the description using Google Translate. / 不幸的是，团队中没有人会说中文，但我想我使用谷歌翻译理解了描述。

There was one change in the latest update that affected lights: lights that were set to be hidden in-game still emitted light. One earlier report about the issue: https://github.com/FakeFishGames/Barotrauma/discussions/13806

I wonder if that could be what's happening in your case too, are the lights set to be hidden in-game? If not, would it be possible to send us the submarine file here as a .zip so we can take a look and see if we can find the problem? Thanks and sorry for the inconvenience!

**Reply by dishfish** [reply]
Thank you for your help. I did set the lights to hidden, and I will try to solve this problem based on the information you provided

**Reply by dishfish** [reply]
I think I have solved the problem. I found that once the lights are set to hidden state in the submarine editor, they cannot be used when the submarine is loaded. Unsetting the hidden state can solve this problem
